lib/nxdk/format.h

```diff
@@ -0,0 +1,27 @@
+#ifndef __NXDK_FORMAT_H__
+#define __NXDK_FORMAT_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef __cplusplus
+#include <stdbool.h>
+#endif
+
+#include <stdint.h>
+
+/**
+ * Formats a volume with FATX.
```

Comment:
It only does "Quick-Format", as it does not wipe the disk but just initialises a new FATX superblock, chain, root dir. The old data is technically still there. Maybe that would be good to know?

Reply:
Why would a user expect the data to not still be there? It's just a formatting function, and not meant for wiping disks.

The distinction between Quick Format and Full Format comes from a time when HDDs were a lot dumber than today, and didn't manage bad sectors themselves, so a Full Format scanned for bad sectors and marked them as such in the cluster table. It doesn't really make sense to use a "Full Format" on a HDD produces in the last ~25 years, and the terminology wasn't ever really embraced outside of MS.